alacritty.yml

```diff
@@ -25,6 +25,8 @@
   #
   # Specified in number of columns/lines, not pixels.
   # If both are `0`, this setting is ignored.
+  # In order to support wide characters, when the number of columns is set to 1,
+  # we will create a terminal with 2 columns.
```

Comment:
Would it be worth making this an error, instead documenting the fact that 1 = 2 in this case?

Reply:
I like this idea. So would the expected effect be that, a terminal window of the default size is created, and an error saying alacritty only supports minimum 2x1 is shown in it?

Reply:
Yep that would be my assumption, since in general when a config value is an error it falls back to the default.

I'm also noticing that providing only one of either `columns` or `lines` currently falls back to the defaults currently. I don't think this is an issue with this PR, but perhaps is worth opening a separate issue/PR to fix.

Reply:
> making this an error, instead documenting the fact that 1 = 2
@kchibisov @chrisduerr Any additional comments? If no one is against this I'll start working on it.

Reply:
I think that it's a great idea, though I don't care much about the column 1/2 thing. If someone specifies a column 1 and it's 2, that's not the end of the world. And nobody's ever going to do that anyways.

Since there's a really good chance that it would make things significantly more complicated, handling just the zero case makes sense to me. That should just work automatically with std's non-zero types.

Reply:
> handling just the zero case

What does this mean? Currently when either height or width is 0 we are using the default size and I don't think I changed this.

Reply:
I mean that you can easily output an error for 1x1 by using the correct types. But for handling 1x2 you'd have to write a custom deserializer. But I personally don't really mind that one case.

Reply:
I'm also wondering if given the logic behind `should_set_window_size`, should we give an error if only one out of the two of `lines` or `columns` is set?

Reply:
> should we give an error if only one out of the two of lines or columns is set
I think sometimes it makes sense to only set one of the dimensions and let the window manager to set the other, though I cannot give a concrete example right now... I think you can open another issue to discuss this.
> easily output an error for 1x1 by using the correct types
Can't see how this can be easily done, std has non-zero types but not non-one types...
I'm thinking about implementing this by adding a `validate_config_values` function in `alacritty::config`, which sets the dimensions to the default values if column is set to 1 and uses the `error!` macro to tell the user the problem. Using a custom deserializer won't work when the user is setting the dimensions with command line options.